auth/decorators.py

```diff
@@ -57,6 +58,10 @@ def wrapper(*args, **kwargs):
                 if result.error_message is not None:
                     # Log the failure.
                     authentication_count.labels(result.kind, False).inc()
+                    # Do we only need to abort for JWT based errors?
+                    if result.kind == AuthKind.ssojwt:
+                        abort(401, message=result.error_message)
```

Comment:
Where is this actually being validated? I don't see changes made to the existing cookie decorator.

Reply:
This is used in https://github.com/quay/quay/blob/master/auth/decorators.py#L72 which gets added to all API calls in https://github.com/quay/quay/blob/master/endpoints/api/__init__.py#L72

Reply:
This decorator get's called on the oauth flow and this oauth token is not in the cookie but rather passed as a header. hence validation is here

Reply:
What are we validating it against? Normally, we would compare the token against the one generated in the database. In this case, is it just a matter of whether the signature is valid or not?

Reply:
summarizing the discussion here. I'm passing `verify_signature: True` to the `ODICLoginService` which causes it to verify the jwt token with the public key from the SSO sever

Reply:
ack. did not see  the that it was verifying against the jwks.